### PR TITLE
feat(DC): add a new resource to manage hosted connected

### PR DESCRIPTION
**Pull request by hmbig2**
**What this PR does / why we need it**:

**Which issue this PR fixes**:
*(optional, in `fixes #<issue number>(, fixes #<issue_number>, ...)` format, will close that issue when PR gets merged)*
fixes #xxx

**Special notes for your reviewer**:

😞 😞 😞 This resource needs other user's tenant_id and host_id, So skip the acceptance test.

**Release note**:

```release-note

```

## PR Checklist

* [X] Tests added/passed.
* [X] Documentation updated.
* [X] Schema updated.

## Acceptance Steps Performed

```
Running tool: /usr/local/go/bin/go test -timeout 9000000000s -run ^TestAccHostedConnect_basic$ github.com/huaweicloud/terraform-provider-huaweicloud/huaweicloud/services/acceptance/dc -v

=== RUN   TestAccHostedConnect_basic
=== PAUSE TestAccHostedConnect_basic
=== CONT  TestAccHostedConnect_basic
    /home/huawei/git/terraform-provider-huaweicloud/huaweicloud/services/acceptance/dc/acceptance.go:833: HW_DC_RESOURCE_TENANT_ID, HW_DC_HOSTTING_ID must be set for this acceptance test
--- SKIP: TestAccHostedConnect_basic (0.00s)
PASS
ok  	github.com/huaweicloud/terraform-provider-huaweicloud/huaweicloud/services/acceptance/dc	0.037s

```

### CODE REVIEW

File: docs/resources/dc_hosted_connect.md [hunk 1]
```diff
@@ -0,0 +1,84 @@
...
+
+* `status` - The status of the hosted connect.  
+  The options are as follows:
+    + **BUILD**: The hosted connect has been created.
+    + **ACTIVE**: The associated virtual gateway is normal.
+    + **DOWN**: The port used by the hosted connect is down, indicating that there may be line faults.
+    + **ERROR**: The associated virtual gateway is abnormal.
+    + **PENDING_DELETE**: The hosted connect is being deleted.
+    + **PENDING_UPDATE**: The hosted connect is being updated.
+    + **PENDING_CREATE**: The hosted connect is being created.
```
Comment (Lance52259) [inline docs/resources/dc_hosted_connect.md]: two spaces we recommend

File: docs/resources/dc_hosted_connect.md [hunk 2]
```diff
@@ -0,0 +1,84 @@
+---
+subcategory: "Direct Connect (DC)"
+---
+
+# huaweicloud_dc_hosted_connect
+
+Manages a hosted connection resource within HuaweiCloud.
+The creator must have the partner qualification and have an operations connection.
```
Comment (ShiChangkuo) [inline docs/resources/dc_hosted_connect.md]: suggest to add a Note
```suggestion
Manages a hosted connection resource within HuaweiCloud.

-> The creator **must** have the partner qualification and have an operations connection.
```

File: huaweicloud/services/dc/resource_huaweicloud_dc_hosted_connect.go [hunk 3]
```diff
@@ -0,0 +1,447 @@
...
+			"resource_tenant_id": {
+				Type:        schema.TypeString,
+				Required:    true,
+				ForceNew:    true,
+				Description: `The tenant ID for whom a hosted connect is to be created.`,
+			},
+			"description": {
+				Type:        schema.TypeString,
+				Optional:    true,
+				Computed:    true,
```
Comment (ShiChangkuo) [inline huaweicloud/services/dc/resource_huaweicloud_dc_hosted_connect.go]: `Computed` should be deleted for description

File: docs/resources/dc_hosted_connect.md [hunk 4]
```diff
@@ -0,0 +1,84 @@
...
+variable hosting_id {}
+
+resource "huaweicloud_dc_hosted_connect" "test" {
+   name               = "demo"
+   description        = "This is a demo"
+   resource_tenant_id = var.resource_tenant_id
+   hosting_id         = var.hosting_id
+   vlan               = 441
+   bandwidth          = 10
+}
```
Comment (ShiChangkuo) [inline docs/resources/dc_hosted_connect.md]: format error

```suggestion
resource "huaweicloud_dc_hosted_connect" "test" {
  name               = "demo"
  description        = "This is a demo"
  resource_tenant_id = var.resource_tenant_id
  hosting_id         = var.hosting_id
  vlan               = 441
  bandwidth          = 10
}
```

File: docs/resources/dc_hosted_connect.md (hunk 2 above)
Comment (hmbig2) [reply docs/resources/dc_hosted_connect.md]: done

File: huaweicloud/services/dc/resource_huaweicloud_dc_hosted_connect.go (hunk 3 above)
Comment (hmbig2) [reply huaweicloud/services/dc/resource_huaweicloud_dc_hosted_connect.go]: done

File: docs/resources/dc_hosted_connect.md (hunk 4 above)
Comment (hmbig2) [reply docs/resources/dc_hosted_connect.md]: done

File: docs/resources/dc_hosted_connect.md (hunk 1 above)
Comment (hmbig2) [reply docs/resources/dc_hosted_connect.md]: done

File: huaweicloud/services/dc/resource_huaweicloud_dc_hosted_connect.go [hunk 5]
```diff
@@ -0,0 +1,446 @@
...
+			"hosting_id":         d.Get("hosting_id"),
+			"vlan":               d.Get("vlan"),
+			"resource_tenant_id": d.Get("resource_tenant_id"),
+			"peer_location":      utils.ValueIngoreEmpty(d.Get("peer_location")),
+		},
+	}
+	return bodyParams
+}
+
+func hostedConnectWaitingForStateCompleted(ctx context.Context, d *schema.ResourceData, meta interface{}, t time.Duration) error {
```
Comment (ShiChangkuo) [inline huaweicloud/services/dc/resource_huaweicloud_dc_hosted_connect.go]: can we reuse the client then it's unnecessary to build the client in **Refresh** func.
```suggestion
func hostedConnectWaitingForStateCompleted(ctx context.Context, client *golangsdk.ServiceClient, id string, t time.Duration) error {
```

File: huaweicloud/services/dc/resource_huaweicloud_dc_hosted_connect.go [hunk 6]
```diff
@@ -0,0 +1,446 @@
...
+	}
+
+	err = deleteHostedConnectWaitingForStateCompleted(ctx, d, meta, d.Timeout(schema.TimeoutDelete))
+	if err != nil {
+		return diag.Errorf("error waiting for the hosted connect (%s) deletion to complete: %s", d.Id(), err)
+	}
+	return nil
+}
+
+func deleteHostedConnectWaitingForStateCompleted(ctx context.Context, d *schema.ResourceData, meta interface{}, t time.Duration) error {
```
Comment (ShiChangkuo) [inline huaweicloud/services/dc/resource_huaweicloud_dc_hosted_connect.go]: ditto

File: huaweicloud/services/dc/resource_huaweicloud_dc_hosted_connect.go (hunk 5 above)
Comment (hmbig2) [reply huaweicloud/services/dc/resource_huaweicloud_dc_hosted_connect.go]: ok，done

File: huaweicloud/services/dc/resource_huaweicloud_dc_hosted_connect.go (hunk 6 above)
Comment (hmbig2) [reply huaweicloud/services/dc/resource_huaweicloud_dc_hosted_connect.go]: done

**Pull request merged**